submariner-operator/move-subctl-to-its-own-repo.md

```diff
@@ -0,0 +1,55 @@
+# Move subctl to its own repository
+
+## Summary
+
+Currently, `subctl` code lies in `submariner-operator` repository. Both subctl and operator require different set of dependencies which
+might conflict will improving subctl/operator. For eg, subctl can't be go install-ed currently because the operator dependencies don't allow
+it. It also makes operator upgrades difficult.
+
+## Proposal
+
+subctl, soon to be a library itself, should be moved out of submariner-operator repositoryto its own repo. Everything specific to the subctl
+is moved to the new subctl repo. subctl repositorydepends on the Operator repo. Operator repositoryshould contain things that the operator
+framework generates/needs.
+
+### Pros
+
+* Operator framework upgrades would be simpler.
+* It's logically correct to separate operator and a library as they are 2 different deliverables and provides different ways to install
+Submariner.
+* Testing criterias are different. We don't need subctl related GHAs for operator and vice versa.
+* Development approach is different. One is a CLI and another is a framework.
+* submariner-addon integration might be simplified.
+
+### Cons
+
+* A new repositoryto maintain.
+
+## Design details
+
+A new GitHub repository `subctl` will be created under `submariner-io` organisation. `submariner-operator` repositorycontent will be copied
+to the new `subctl` repository and then unrelated content would be deleted. This will help in maintaining the history as to what content
+was moved out of `submariner-operator` repository. The `subctl` repository will preserve all the code owner information and admin rights
+from the `submariner-operator` repository. It will also follow
+[standatrd Go project layout](https://github.com/golang-standards/project-layout).
```

Comment:
```suggestion
A new GitHub repository `subctl` will be created under the `submariner-io` organisation.
The `submariner-operator` repository content will be copied to the new `subctl` repository,
by pushing the git repository as-is, so that the full history is preserved.
Non-`subctl`-related content will then be removed from the `subctl` repository, and non-Operator-related content will be removed from the `submariner-operator` repository.
Once this is complete, there should be no duplicate content in the two repositories, other than the build framework.
The `subctl` repository will be set up with the same permissions, branch setup, and code ownership as the `submariner-operator` repository.
The `subctl` repository will also follow
[the standard Go project layout](https://github.com/golang-standards/project-layout).
```